error[E0599]: no method named `embedding_search` found for struct `CueBuilder<engram_core::memory::cue_builder_states::NoId>` in the current scope
  --> tests/compile_fail/cue_builder_invalid_sequence.rs:14:28
   |
14 |     let _builder = builder.embedding_search([0.1f32; 768], Confidence::MEDIUM); //~ ERROR: method `embedding_search` not found for this v...
   |                            ^^^^^^^^^^^^^^^^ method not found in `CueBuilder<engram_core::memory::cue_builder_states::NoId>`
   |
   = note: the method was found for
           - `CueBuilder<NoCueType>`

error[E0599]: no method named `build` found for struct `CueBuilder<NoCueType>` in the current scope
  --> tests/compile_fail/cue_builder_invalid_sequence.rs:24:32
   |
24 |     let _cue = partial_builder.build(); //~ ERROR: method `build` not found for this value
   |                                ^^^^^ method not found in `CueBuilder<NoCueType>`
   |
   = note: the method was found for
           - `CueBuilder<engram_core::memory::cue_builder_states::Ready>`
