error[E0599]: no method named `build` found for struct `CueBuilder<engram_core::memory::cue_builder_states::NoId>` in the current scope
  --> tests/compile_fail/cue_builder_educational_errors.rs:10:34
   |
10 |     let _cue = CueBuilder::new().build(); //~ ERROR: method `build` not found for this value
   |                                  ^^^^^ method not found in `CueBuilder<NoId>`
   |
   = note: the method was found for
           - `CueBuilder<engram_core::memory::cue_builder_states::Ready>`

error[E0599]: no method named `max_results` found for struct `CueBuilder<engram_core::memory::cue_builder_states::NoId>` in the current scope
  --> tests/compile_fail/cue_builder_educational_errors.rs:17:25
   |
17 |     let _fail = builder.max_results(10); //~ ERROR: method `max_results` not found for this value
   |                         ^^^^^^^^^^^ private field, not a method

error[E0599]: no method named `cue_confidence` found for struct `CueBuilder<NoCueType>` in the current scope
  --> tests/compile_fail/cue_builder_educational_errors.rs:25:28
   |
25 |     let _invalid = partial.cue_confidence(engram_core::Confidence::HIGH); //~ ERROR: method `cue_confidence` not found for this value
   |                            ^^^^^^^^^^^^^^ private field, not a method
