warning: unused import: `engram_core::Confidence`
 --> tests/compile_fail/memory_builder_invalid_sequence.rs:7:5
  |
7 | use engram_core::Confidence;
  |     ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0599]: no method named `embedding` found for struct `MemoryBuilder<engram_core::memory::memory_builder_states::NoId>` in the current scope
  --> tests/compile_fail/memory_builder_invalid_sequence.rs:14:28
   |
14 |     let _builder = builder.embedding([0.1f32; 768]); //~ ERROR: method `embedding` not found for this value
   |                            ^^^^^^^^^ private field, not a method

error[E0599]: no method named `build` found for struct `MemoryBuilder<engram_core::memory::memory_builder_states::NoEmbedding>` in the current scope
  --> tests/compile_fail/memory_builder_invalid_sequence.rs:24:35
   |
24 |     let _memory = partial_builder.build(); //~ ERROR: method `build` not found for this value
   |                                   ^^^^^ method not found in `MemoryBuilder<engram_core::memory::memory_builder_states::NoEmbedding>`
   |
   = note: the method was found for
           - `MemoryBuilder<engram_core::memory::memory_builder_states::Ready>`
